docs/bootstrap.json

```diff
@@ -0,0 +1,20 @@
+{
+  "ID": "bootstrap",
+  "Properties": {
+    "Resources": {
+      "resource_a": {
+        "Plugin": "resource-file",
+        "Type": "a",
+        "Properties": {
+          "Note": "Depends on {{.Resources.resource_b.ID}}"
```

Comment:
This reference here, `.Resource.resource_b.ID`, isn't explicit -- in the sense that I can't find anywhere in the document that says it satisifies the contract of having a field called `ID`.   This can be a problem in that it's impossible to validate this input (e.g, is it `ID` or maybe it's `Id`?)  

Another concern here is that this expression `.Resources.resource_b.ID` is closely coupled to the *structure* of this document and if there's a shuffling in schema all the expressions will have to updated to reflect that.  

Let me know if you agree.  I have some ideas how we can mitigate this.   
```